src/extensions/timeseries/TimeseriesViewNew/TimeseriesWidgetNew.tsx

```diff
@@ -160,8 +161,9 @@ class Panel {
 }
 
 const TimeseriesWidgetNew = (props: Props) => {
-    const { timeseriesData, width, height, y_scale_factor, channel_ids, visibleChannelIds, recordingSelection, recordingSelectionDispatch } = props
+    const { timeseriesData, width, height, y_scale_factor, channel_ids, visibleChannelIds, recordingSelection: externalSelection, recordingSelectionDispatch: externalSelectionDispatch } = props
     const [panels, setPanels] = useState<Panel[]>([])
+    const [recordingSelection, recordingSelectionDispatch] = useBufferedDispatch(recordingSelectionReducer, externalSelection, useMemo(() => ((state: RecordingSelection) => {externalSelectionDispatch({type: 'Set', state})}), [externalSelectionDispatch]), 400)
```

Comment:
useBufferedDispatch improves the responsiveness of the widget. Actions are immediately applied, and dispatch to the larger app is delayed.